File: Java/Easy/WrongAnswer/minimum-amount-of-time-to-fill-cups_9.java
class Solution {
    public int fillCups(int[] amount) {
        
        int ans = 0;
        for(int i=0; i<3; i++){
            ans += amount[i];
        }
        int[] arr = Arrays.copyOf(amount, 3);
        if(arr[0] > arr[2]){
            int temp = arr[0];
            arr[0] = arr[2];
            arr[2] = temp;
        }
        if(arr[1] > arr[2]){
            int temp = arr[1];
            arr[1] = arr[2];
            arr[2] = temp;
        }
        int min = Math.min(arr[0], arr[1]);
        ans += min;
        arr[0] -= min;
        arr[1] -= min;
        ans += arr[2];
        return ans;
    }
}

File: C++/Easy/WrongAnswer/minimum-amount-of-time-to-fill-cups_9.cpp
class Solution {
public:
    int fillCups(vector<int>& amount) {
        
        int n = amount.size();
        int res = 0;
        int i = 0;
        while (i < n) {
            if (amount[i] > 0) {
                amount[i]--;
                res++;
            }
            i++;
        }
        int j = 0;
        while (j < n) {
            if (amount[j] > 0) {
                amount[j]--;
                res++;
            }
            j++;
        }
        int k = 0;
        while (k < n) {
            if (amount[k] > 0) {
                amount[k]--;
                res++;
            }
            k++;
        }
        return res;
    }
    int minTimeToFill(vector<int>& amount) {
        int res = 0;
        int n = amount.size();
        int i = 0;
        while (i < n) {
            if (amount[i] > 0) {
                res = fillCups(amount);
                break;
            }
            i++;
        }
        return res;
    }
};